File: src/main/java/org/cubewhy/celestial/game/addon/FabricMod.java
/*
 * Celestial Launcher <me@lunarclient.top>
 * License under GPLv3
 * Do NOT remove this note if you want to copy this file.
 */

package org.cubewhy.celestial.game.addon;

import org.cubewhy.celestial.Celestial;
import org.cubewhy.celestial.game.BaseAddon;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.io.File;
import java.io.IOException;
import java.util.ArrayList;
import java.util.List;
import java.util.Objects;

public class FabricMod extends BaseAddon {
    public static final File modFolder = new File(Celestial.config.getValue("installation-dir").getAsString(), "mods");
    public final File file;

    public FabricMod(File file) {
        this.file = file;
    }


    public static @NotNull List<FabricMod> findAll() {
        List<FabricMod> list = new ArrayList<>();
        if (modFolder.isDirectory()) {
            for (File file : Objects.requireNonNull(modFolder.listFiles())) {
                if (file.getName().endsWith(".jar") && file.isFile()) {
                    list.add(new FabricMod(file));
                }
            }
        }
        return list;
    }

    public static @Nullable FabricMod add(File file) throws IOException {
        File target = autoCopy(file, modFolder);
        return (target == null) ? null : new FabricMod(target);
    }

    @Override
    public String toString() {
        return this.file.getName();
    }
}

File: src/main/kotlin/org/cubewhy/celestial/game/addon/FabricMod.kt
/*
 * Celestial Launcher <me@lunarclient.top>
 * License under GPLv3
 * Do NOT remove this note if you want to copy this file.
 */
package org.cubewhy.celestial.game.addon

import org.cubewhy.celestial.config
import org.cubewhy.celestial.game.BaseAddon
import org.cubewhy.celestial.toFile
import java.io.File
import java.util.*

class FabricMod(val file: File) : BaseAddon() {
    override fun toString(): String {
        return file.name
    }

    override val isEnabled: Boolean
        get() =// TODO fabric: isEnabled
            true

    override fun toggle(): Boolean {
        return toggle(file)
    }

    companion object {
        val modFolder: File = config.installationDir.toFile().resolve("mods")


        fun findAll(): List<FabricMod> {
            val list: MutableList<FabricMod> = ArrayList()
            if (modFolder.isDirectory) {
                for (file in Objects.requireNonNull<Array<File>>(modFolder.listFiles())) {
                    if (file.name.endsWith(".jar") && file.isFile) {
                        list.add(FabricMod(file))
                    }
                }
            }
            return list
        }


        fun add(file: File?): FabricMod? {
            val target = autoCopy(file!!, modFolder)
            return if ((target == null)) null else FabricMod(target)
        }
    }
}
